km51/Nikolaiev_Dmytro/laboratory1.sql

```diff
@@ -1,2 +1,139 @@
 -- LABORATORY WORK 1
 -- BY Nikolaiev_Dmytro
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+створювати таблиці
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+Create User Nikolaev identified by nikolaev;
+Grant CREATE ANY TABLE to Nikoalev;
+Create Role "Connect";
+Grant "Connect" to Nikolaev;
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Людина співає пісню.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE table humans (
+human_name varchar2 NOT NULL
+);
+
+CREATE table songs (
+song_name varchar2 NOT NULL
+);
+
+ALTER TABLE humans 
+ADD Constraint human_pk Primary key (human_name);
+
+Alter table songs 
+ADD Constraint song_pk Primary Key (song_name);
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+GRANT Create Any table, select any table, insert any table to Nikoalev;
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Скільки проданого найдорожчого товару?
+Виконати завдання в SQL. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+SELECT Count(prod_price)
+FROM products 
+where prod.price=Max(prod.price);
+
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.b. 
+Який PROD_ID товару, з найкоротшою назвою?
+Виконати завдання в SQL. 
+4 бали
+
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+Select DISTINCT (prod_id)
+From products
+where prod_id=Min(prod_id);
+
```

Comment:
You can`t do min from string, you should recall trim function and length function.
```
project prod_id times products
where length(trim(prod_name)) = (project min(length(trim(prod_name))) times products)
```